src/cordova-plugin-outline/package.json

```diff
@@ -9,4 +9,4 @@
       "osx"
     ]
   }
-}
\ No newline at end of file
+}
```

Comment:
Can we call the directory simply `cordova_plugin/`?

We don't otherwise use dashes in file names and outline is redundant.

Reply:
Actually, I think the plugin should go under src/cordova/plugin/, so the cordova code stays in one place.

Reply:
So it seems that the package.json is required: in anticipation of turning this into a monorepo, I advise that we store the plugin in `src/cordova_plugin` - that alright?

Reply:
still works: 
<img width="408" alt="Screen Shot 2022-11-08 at 13 45 37" src="https://user-images.githubusercontent.com/3759828/200649364-d87c5812-b7b9-4b7a-8d41-45643fc6077e.png">

Reply:
I don't understand. Why can we have `cordova-plugin-outline/package.json`, but not `src/cordova/plugin/package.json`? Nothing other than root has a package.json.

On your future concerns, perhaps "src/cordova" shouldn't be a package? That's not a component we build, it's really just a folder for organizing the code. In any case, it would be easy to move later.

Reply:
The thing is, if we make "src/cordova" a package, it will allow us to scope the artifacts (i.e. platforms, plugins, www) within that package.json-containing folder rather than having them live in root

We can have `src/cordova/plugin/package.json` it's just not a clean, predictable monorepo that way. AFAIK you can't nest workspaces (the best scenario would be to have a "cordova workspace" within the larger "outline-client" workspace)

Reply:
To me it's a requirement to be able to place packages at any directory level, so we can properly organize them. Having them be at root limits how we can organize things.

Avoiding nested packages is fine, we can work around that.

I don't get it: why do you want to make src/cordova a package? Can you clarify "allow us to scope the artifacts"?

If you need a package for the cordova app, then perhaps we need src/cordova/app?

In the end we will probably have separate directories per platforms anyway if we move to platform-centered development.

Reply:
As a note about the nesting, it's better if we don't need to see the cordova plugin dir if we're only dealing with electron, for example. Flattening exposes complexity, and loses valuable semantic information.

Reply:
I outlined some options in this doc: https://docs.google.com/document/d/1s9Ij9HEVt-5G1PhGWZbmWG3_or95ASp9T2nXkSVMVng/edit?resourcekey=0-LMg3pZDaEl68DdC3uQLWZw

Reply:
> In the end we will probably have separate directories per platforms anyway if we move to platform-centered development.

I think the best way to do this is to have a separate `package.json` per platform. I think that way we can avoid nesting workspaces and keep semantic foldering, there may just be some repeated dependency declarations, if that's okay.

Reply:
Maybe later we name src to "client" to support a sister folder for "manager" or something else.

Reply:
BTW, we have many components, not really one per platform. We have installers, services, etc. Each of them is probably a package. We should better represent them at some point. I tried to list some of them at https://docs.google.com/document/d/1PMrY3U0DbxggdmlJTIewWS_kcykudbCMc0RzfOalpkc/edit#heading=h.5z1c56ucp658.

Reply:
Yeah, fair point. We can take them one at a time.